### PR TITLE
 [Project A][Win11][macareonie] Upgrade to Java 17

### PR DESCRIPTION
**Pull request by macareonie**
# Migrate to Java 17

## Description
This PR migrates the project from Java 11 to Java 17, maintaining back compatibility with Java 11 and resolving warnings generated from code using versions below Java 11.

## Details
- **OS used:** Windows 11
- **JDK distribution used:** Oracle OpenJDK version 17.0.11
- **IDE used:** IntelliJ IDEA
- **Link to JAR file:** [Download here](https://github.com/macareonie/AB3-J17/releases/tag/v1.0)

## Other info
The warnings from previous Java versions were resolved by implementing changes suggested by the IDE.

### CODE REVIEW

**Comment by itstrueitstrueitsrealitsreal** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

### Steps followed:
1. Downloaded the JAR file from the PR.
2. Switched to JDK-FX 17.0.11.fx-zulu using SDKMAN.
3. Ran the JAR file using `java -jar Win11-macareonie.jar`.
4. Performed a smoke test to verify basic functionality.

### Results:
Everything is functioning as expected.

### Screenshots:
Screenshot 2024-07-04 at 4 42 46 PM

**Comment by jyue487** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

Steps and results are identical to above.

**Comment by teoks0199** [comment]
Tested on Windows 11, and working.

Steps followed:

1. Downloaded the JAR file from the PR.
2. Ran the JAR file using java -jar Win11-macareonie.jar.
3. Performed a smoke test to verify basic functionality.

image

**Comment by bennyLCK** [comment]
Smoke-Tested on Windows 11, using Oracle OpenJDK version 17.0.10 on Windows Powershell Terminal same as @teoks0199, basic commands were run to ensure full functionality of the application.

image

**Comment by baskargopinath** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

Steps followed:
Downloaded the JAR file from the PR.
Switched to JDK-FX 17.0.11.fx-zulu
Ran the JAR file using java -jar Win11-macareonie.jar.
Performed a smoke test to verify basic functionality.
Results:
Everything is functioning as expected.

Screenshots:
image

**Comment by aureliony** [comment]
Tested on Ubuntu 22.04.4 LTS with zulu jdk 17.0.11. JAR is working fine.

Steps followed:
Downloaded the JAR file from the release.
Ran the JAR file using `java -jar Win11-macareonie.jar`.
Performed a smoke test to check basic functionality.

Results:
Everything is functioning as expected.
Screenshot from 2024-07-06 19-37-14

**Comment by gongg21** [comment]
Tested on **MacSilicon** with JDK-FX 17.0.11.fx-zulu, and working fine.
Switched using SDKMAN, ensure it is **17.0.11.fx-zulu**, note the `.fx` suffix.
Add/Delete command tested, works.
Help command tested, works.
Ran JAR file using `java -jar <filename>`.

Screenshot 2024-07-07 at 5 38 32 PM

**Comment by Carlintyj** [comment]
Tested on Mac (Silicon) using JDK-FX 17.0.11.fx-zulu and working.

Downloaded the JAR release, ran the jar file using java -jar Win11-macareonie.jar
Performed smoke test to check functionality.

Screenshot 2024-07-07 at 7 18 53 PM

**Comment by jedkohjk** [comment]
Tested on Windows 11 and working.

Steps I followed:
1. Downloaded the JAR file from the release.
2. Ran `java -jar Win11-macareonie.jar`.
3. Tested some basic commands.

Screenshot

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java [hunk 1]
```diff
@@ -87,7 +88,7 @@ private static int findPrefixPosition(String argsString, String prefix, int from
     private static ArgumentMultimap extractArguments(String argsString, List<PrefixPosition> prefixPositions) {
 
         // Sort by start position
-        prefixPositions.sort((prefix1, prefix2) -> prefix1.getStartPosition() - prefix2.getStartPosition());
+        prefixPositions.sort(Comparator.comparingInt(PrefixPosition::getStartPosition));
```
Comment (damithc) [inline src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:91]: @macareonie Some of these changes are not strictly needed for the Java17 migration. If you think they are useful to have nevertheless, you can send separate PRs for them.

**Comment by damithc** [comment]
Closing, as the migration was done in https://github.com/nus-oss/AB3-J17/commit/fe805d8553bbc329644371a6e732a8317ea6686f
